dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/DataOkHttpUploaderV2.kt

```diff
@@ -34,17 +34,18 @@ internal abstract class DataOkHttpUploaderV2(
 
     private val uploaderName = javaClass.simpleName
 
+    internal val clientToken = if (isValidHeaderValue(rawClientToken)) rawClientToken else ""
+    internal val source: String = sanitizeHeaderValue(rawSource)
+    internal val sdkVersion: String = sanitizeHeaderValue(rawSdkVersion)
+
     private val userAgent by lazy {
-        System.getProperty(SYSTEM_UA).let {
-            if (it.isNullOrBlank()) {
-                "Datadog/$sdkVersion " +
+        sanitizeHeaderValue(System.getProperty(SYSTEM_UA))
```

Comment:
Mostly because the syntax for User Agent can become very sketchy and more likely to have weird characters. 
Also the `Build.xxx` will be already fetched from the app zygote, whereas the UA is retrieved at Runtime, making it possible for the host app to override it.

Reply:
Got your point. It is a good trade-off.